kb/0003-scripting-functions.md

```diff
@@ -5,7 +5,711 @@ _[back to Index](../index.md)_
+    ```
+
+* **GetAttributeName**
+    * **Compiler Token**: `FUNC_GET_ATTRIBUTE_NAME`
+    ``` C++
+    syntax:
+        string GetAttributeName(object obj);
+        string GetAttributeName(aref attribute);
+    ```
+
+* **DelEventHandler**
+    * **Compiler Token**: `FUNC_DEL_EVENT_HANDLER`
+    ``` C++
+    syntax:
+        void DelEventHandler(string eventName, string functionName);
+    ```
+
+* **EntityUpdate**
+    * **Compiler Token**: `FUNC_Entity_UPDATE`
+    ``` C++
+    syntax:
+        void EntityUpdate(bool isEnabled);
+    ```
+
+* **IsEntity**
+    * **Compiler Token**: `FUNC_IS_Entity_LOADED`
+    ``` C++
+    syntax:
+        int IsEntity(object obj);
+    ```
+
+* **DumpAttributes**
+    * **Compiler Token**: `FUNC_DUMP_ATTRIBUTES`
+    ``` C++
+    syntax:
+        void DumpAttributes(ref objRef);
+    ```
+
+* **sti**
+    * **Compiler Token**: `FUNC_STI`
+    ``` C++
+    syntax:
+        int sti(string value);
+    ```
+
+* **stf**
+    * **Compiler Token**: `FUNC_STF`
+    ``` C++
+    syntax:
+        float stf(string value);
+    ```
+
+* **CheckAttribute**
+    * **Compiler Token**: `FUNC_CHECK_ATTRIBUTE`
+    ``` C++
+    syntax:
+        bool CheckAttribute(object &obj, string attribute);
+    ```
+
+* **argb**
+    * **Compiler Token**: `FUNC_ARGB`
+    ``` C++
+    syntax:
+        int argb(int a, int r, int g, int b);
+    ```
+
+* **DeleteEntities**
+    * **Compiler Token**: `FUNC_DELETE_ENTITIES`
+    ``` C++
+    syntax:
+        void DeleteEntities();
+    ```
+
+* **ClearEvents**
+    * **Compiler Token**: `FUNC_CLEAR_EVENTS`
+    ``` C++
+    syntax:
+        void ClearEvents(); 
+    ```
+
+* **SaveEngineState**
+    * **Compiler Token**: `FUNC_SAVEENGINESTATE`
+    ``` C++
+    syntax:
+        void SaveEngineState(1);
+    ```
+
+* **LoadEngineState**
+    * **Compiler Token**: `FUNC_LOADENGINESTATE`
+    ``` C++
+    syntax:
+        void LoadEngineState(); /1/?
+    ```
+
+* **Event**
+    * **Compiler Token**: `FUNC_EVENT`
+    ``` C++
+    syntax:
+        void Event(); 
+    ```
+
+* **PostEvent**
+    * **Compiler Token**: `FUNC_POSTEVENT`
+    ``` C++
+    syntax:
+        void PostEvent(); 
+    ```
+
+* **fts**
+    * **Compiler Token**: `FUNC_FTS`
+    ``` C++
+    syntax:
+        string fts(float, int); 
+    ```
+
+* **ClearPostEvents**
+    * **Compiler Token**: `FUNC_CLEAR_POST_EVENTS`
+    ``` C++
+    syntax:
+        void ClearPostEvents();
+    ```
+
+* **SetArraySize**
+    * **Compiler Token**: `FUNC_SET_ARRAY_SIZE`
+    ``` C++
+    syntax:
+        void SetArraySize(2); 
+    ```
+
+* **GetAttributeValue**
+    * **Compiler Token**: `FUNC_GET_ATTRIBUTE_VALUE`
+    ``` C++
+    syntax:
+        string GetAttributeValue(aref attribute);
+    ```
+
+* **Vartype**
+    * **Compiler Token**: `FUNC_VARTYPE`
+    ``` C++
+    syntax:
+        string Vartype(aref attribute);
+
+    /* TODO(yakvi): actually confirm it's an aref input, from the source looks like it 
```

Comment:
@mitrokosta, could you confirm it?

Reply:
Vartype was never used in scripts, therefore I can't confirm or dispute it based on script usage. From the source to me it looks like it's actually a ref input, not aref.